metecho/api/models.py

```diff
@@ -451,6 +453,60 @@ def get_head(self):
 
     # end CreatePrMixin configuration
 
+    def create_gh_branch(self):
+        if not self.id:  # i.e. "newly created"
+            # We cannot get an app user with a repo_id, sadly; the API
+            # doesn't support it. So we have to use the repo_owner and
+            # repo_name.
+            repository = gh.get_repo_info(
+                None,
+                repo_owner=self.repository.repo_owner,
+                repo_name=self.repository.repo_name,
+            )
+
+            if self.branch_name:
+                try:
+                    head = repository.branch(self.branch_name).commit.sha
+                except NotFoundError:
+                    gh.try_to_make_branch(
+                        repository,
+                        new_branch=self.branch_name,
+                        base_branch=repository.default_branch,
+                    )
+                else:
+                    base = repository.branch(repository.default_branch).commit.sha
+                    self.has_unmerged_commits = (
+                        "ahead" in repository.compare_commits(base, head).status
+                    )
+                    # Check if has PR
+                    try:
+                        head_str = f"{repository.owner}:{self.branch_name}"
+                        base_str = f"{repository.owner}:{repository.default_branch}"
+                        # Defaults to descending order, so we'll find
+                        # the most recent one, if there is one to be
+                        # found:
+                        pr = next(
+                            repository.pull_requests(head=head_str, base=base_str)
+                        )
+                        # Check PR status
+                        self.pr_number = pr.number
+                        self.pr_is_open = pr.closed_at is None
+                        self.pr_is_merged = pr.is_merged
+                    except StopIteration:
+                        pass
+            else:
+                from .jobs import project_create_branch
+
+                repo_id = self.get_repo_id(None)
+                project_create_branch(
+                    project=self,
+                    repository=repository,
+                    # This is used only to check out the repo to get config info:
+                    user=None,
```

Comment:
@wlonk Wait... if `user` isn't required by `project_create_branch`, why do we pass it in at all?